korman/operators/op_world.py

```diff
@@ -14,14 +14,78 @@
 #    along with Korman.  If not, see <http://www.gnu.org/licenses/>.
 
 import bpy
-
+from bpy.props import *
+from pathlib import Path
 
 class AgeOperator:
     @classmethod
     def poll(cls, context):
         return context.scene.render.engine == "PLASMA_GAME"
 
 
+class GameAddOperator(AgeOperator, bpy.types.Operator):
+    bl_idname = "world.plasma_game_add"
+    bl_label = "Add Plasma Game"
+
+    filepath = StringProperty(subtype="DIR_PATH")
+    directory = BoolProperty(default=True, options={"HIDDEN"})
+    game_index = IntProperty(default=-1, options={"HIDDEN"})
+
+    def execute(self, context):
+        w = context.world
+        if w:
+            # First, verify this is a valid Uru directory...
+            path = Path(self.filepath)
+            if not path.is_dir():
```

Comment:
Unfortunately Blender likes to auto-fill the filename field in the selector dialog, so we should strip that portion to avoid confounding error messages when the user selected a directory and Blender is too "helpful".